### PR TITLE
SL-639: forward ref to textarea

### PR DESCRIPTION
**Pull request by wmhilton**
To do some of the fancy keyboard navigation needed for the Slate editor, we need access to the underlying textarea component, so I am forwarding a ref.

### CODE REVIEW

File: src/CodeEditor.tsx [hunk 1]
```diff
@@ -49,17 +49,17 @@ export const supportedLanguages = {
   ...optionalSupport,
 };
 
-const CodeEditorView = (props: ICodeEditorProps & { className: string }) => {
+const CodeEditorView = React.forwardRef((props: ICodeEditorProps & { className: string }, ref: any) => {
```
Comment (P0lip) [inline src/CodeEditor.tsx]: Let's add some typing here.
```ts
React.forwardRef<ReactSimpleCodeEditor, ICodeEditorProps & { className: string }>((props, ref) => {
```

Comment (wmhilton) [reply src/CodeEditor.tsx]: Why? TS seems to do a good job infering it automatically.
image

Edit: Ooooh I see, it _doesnt_ do a good job inferring it. `ForwardRefExoticComponent` is not as specific as `ReactSimpleCodeEditor`. OK, I got this!

Comment (wmhilton) [reply src/CodeEditor.tsx]: Weirdly, it seems like TS still sees the same type:
image

Comment (P0lip) [reply src/CodeEditor.tsx]: It does not seem to be exactly the same one. See the lack of ReactSimpleCodeEditor in RefAttributes in the first screenshot.

Comment (wmhilton) [reply src/CodeEditor.tsx]: I'm nearsighted... I only see the first 3 lines of any TypeScript type. :)

**Pull request merged**